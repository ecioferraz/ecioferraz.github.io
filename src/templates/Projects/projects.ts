import articleSearch from '../../../public/assets/images/projects/articleSearch.png';
import myTunes from '../../../public/assets/images/projects/myTunes.png';
import ngcash from '../../../public/assets/images/projects/ngcash.png';
import pixelArt from '../../../public/assets/images/projects/pixel-art.png';
import spacetime from '../../../public/assets/images/projects/spacetime.png';
import starWars from '../../../public/assets/images/projects/star-wars.png';
import trivia from '../../../public/assets/images/projects/trivia.png';
import trybewallet from '../../../public/assets/images/projects/trybewallet.png';

export default [
  {
    imgSrc: spacetime,
    mainURL: 'https://github.com/ecioferraz/rocketseat-nlw-ignite',
    secondaryURL: 'https://github.com/ecioferraz/rocketseat-nlw-ignite',
    title: 'Spacetime'
  },
  {
    imgSrc: ngcash,
    mainURL: 'https://github.com/ecioferraz/ngcash-frontend',
    secondaryURL: 'https://github.com/ecioferraz/ngcash-backend',
    title: 'NG.CASH',
  },
  {
    imgSrc: myTunes,
    mainURL: 'https://ecioferraz.github.io/mytunes/',
    secondaryURL: 'https://github.com/ecioferraz/mytunes',
    title: 'myTunes',
  },
  {
    imgSrc: articleSearch,
    mainURL: 'https://ecioferraz.github.io/mettzer-article-search/',
    secondaryURL: 'https://github.com/ecioferraz/mettzer-article-search',
    title: 'Article Search',
  },
  {
    imgSrc: trybewallet,
    mainURL: 'https://ecioferraz.github.io/trybe-trybewallet-project/',
    secondaryURL: 'https://github.com/ecioferraz/trybe-trybewallet-project',
    title: 'TrybeWallet',
  },
  {
    imgSrc: trivia,
    mainURL: 'https://ecioferraz.github.io/trybe-trivia-project/',
    secondaryURL: 'https://github.com/ecioferraz/trybe-trivia-project',
    title: 'Trivia',
  },
  {
    imgSrc: starWars,
    mainURL: 'https://ecioferraz.github.io/trybe-star-wars-project/',
    secondaryURL: 'https://github.com/ecioferraz/trybe-star-wars-project',
    title: 'Star Wars: Planet Search',
  },
  {
    imgSrc: pixelArt,
    mainURL: 'https://ecioferraz.github.io/trybe-pixel-art-project/',
    secondaryURL: 'https://github.com/ecioferraz/trybe-pixel-art-project',
    title: 'Pixel Art',
  },
];
