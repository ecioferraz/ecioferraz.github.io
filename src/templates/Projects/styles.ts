import styled, { css } from 'styled-components';

export const Container = styled.main`
  ${({ theme }) => css`
    @media ${theme.media.height.lteSmall} {
      height: 100%;
    }

    @media ${theme.media.height.lteMedium} {
      h1 {
        font-size: ${theme.font.sizes.large};
      }
    }
  `}
`;

export const ProjectsContainer = styled.section`
    align-items: center;
    display: flex;
    flex-wrap: wrap;
    justify-content: center;
`;
