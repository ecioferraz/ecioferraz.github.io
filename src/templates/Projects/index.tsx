import ProjectCard from '../../components/ProjectCard';
import TextCard from '../../components/TextCard';
import projects from './projects';
import * as Styled from './styles';

export default function Projects() {
  return (
    <Styled.Container id="projects">
      <TextCard>Projetos</TextCard>
      <Styled.ProjectsContainer>
        {projects.map(({ imgSrc, mainURL, secondaryURL, title }) => (
          <ProjectCard
            imgSrc={imgSrc}
            mainURL={mainURL}
            secondaryURL={secondaryURL}
            title={title}
          />
        ))}
      </Styled.ProjectsContainer>
    </Styled.Container>
  );
}
