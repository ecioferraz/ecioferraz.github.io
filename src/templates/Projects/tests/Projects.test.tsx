import renderTheme from '../../../styles/renderTheme';
import Projects from '..';

describe('<Projects />', () => {
  it('should render <Projects />', () => {
    const { container } = renderTheme(<Projects />);

    expect(container).toMatchSnapshot();
  });
});
