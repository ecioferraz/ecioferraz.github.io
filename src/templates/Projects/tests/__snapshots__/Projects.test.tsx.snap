// Vitest Snapshot v1

exports[`<Projects /> > should render <Projects /> 1`] = `
.c8 {
  display: inline-block;
  vertical-align: middle;
  overflow: hidden;
}

.c4 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  margin: 0;
}

.c4 svg {
  color: #000;
  width: 2rem;
}

.c4 svg:hover {
  scale: 1.3;
}

.c5 {
  width: 100%;
}

.c1 {
  font-size: 5.6rem;
}

.c7 {
  font-size: 1.6rem;
}

.c3 {
  background-color: #FFF;
  border: 1px solid rgba(0,0,0,0.125);
  border-radius: 0.8rem;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  margin: 0.8rem;
  max-height: 20rem;
  max-width: 27rem;
  overflow: hidden;
}

.c3,
.c3 * {
  -webkit-transition: 0.8s all ease-in-out;
  transition: 0.8s all ease-in-out;
}

.c3:hover {
  scale: 1.05;
}

.c6 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  color: #000;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex: 1 1 auto;
  -ms-flex: 1 1 auto;
  flex: 1 1 auto;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  padding: 0 1.2rem;
  text-align: center;
}

.c6 h4 {
  font-size: 1.5rem;
  font-weight: 500;
  margin: 0;
  white-space: nowrap;
}

.c2 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

@media (max-width:768px) {
  .c1 {
    font-size: 4.8rem;
  }
}

@media only screen and (max-width:1280px) {
  .c3 {
    max-height: 18rem;
    max-width: 22rem;
  }
}

@media (max-height:540px) {
  .c0 {
    height: 100%;
  }
}

@media (max-height:768px) {
  .c0 h1 {
    font-size: 3.2rem;
  }
}

<div>
  <main
    class="c0"
    id="projects"
  >
    <h1
      class="c1"
    >
      Projetos
    </h1>
    <section
      class="c2"
    >
      <section
        class="c3"
        title="NG.CASH"
      >
        <a
          class="c4"
          href="https://github.com/ecioferraz/ngcash-frontend"
          target="_blank"
        >
          <img
            alt="Imagem do projeto NG.CASH"
            class="c5"
            src="/@fs/home/ecio/Documentos/trybe/Portfólio/ecioferraz.github.io/public/assets/images/projects/ngcash.png"
          />
        </a>
        <section
          class="c6"
        >
          <h4
            class="c7"
          >
            NG.CASH
          </h4>
          <a
            class="c4"
            href="https://github.com/ecioferraz/ngcash-backend"
            target="_blank"
          >
            <svg
              class="c8"
              fill="currentColor"
              focusable="false"
              role="img"
              viewBox="0 0 20 20"
              xmlns="http://www.w3.org/2000/svg"
            >
              <title>
                GitHub
              </title>
              <path
                d="M5.719 14.75a.997.997 0 0 1-.664-.252L-.005 10l5.341-4.748a1 1 0 0 1 1.328 1.495L3.005 10l3.378 3.002a1 1 0 0 1-.664 1.748zm8.945-.002L20.005 10l-5.06-4.498a.999.999 0 1 0-1.328 1.495L16.995 10l-3.659 3.252a1 1 0 0 0 1.328 1.496zm-4.678 1.417 2-12a1 1 0 1 0-1.972-.329l-2 12a1 1 0 1 0 1.972.329z"
              />
            </svg>
          </a>
        </section>
      </section>
      <section
        class="c3"
        title="myTunes"
      >
        <a
          class="c4"
          href="https://ecioferraz.github.io/mytunes/"
          target="_blank"
        >
          <img
            alt="Imagem do projeto myTunes"
            class="c5"
            src="/@fs/home/ecio/Documentos/trybe/Portfólio/ecioferraz.github.io/public/assets/images/projects/myTunes.png"
          />
        </a>
        <section
          class="c6"
        >
          <h4
            class="c7"
          >
            myTunes
          </h4>
          <a
            class="c4"
            href="https://github.com/ecioferraz/mytunes"
            target="_blank"
          >
            <svg
              class="c8"
              fill="currentColor"
              focusable="false"
              role="img"
              viewBox="0 0 20 20"
              xmlns="http://www.w3.org/2000/svg"
            >
              <title>
                GitHub
              </title>
              <path
                d="M5.719 14.75a.997.997 0 0 1-.664-.252L-.005 10l5.341-4.748a1 1 0 0 1 1.328 1.495L3.005 10l3.378 3.002a1 1 0 0 1-.664 1.748zm8.945-.002L20.005 10l-5.06-4.498a.999.999 0 1 0-1.328 1.495L16.995 10l-3.659 3.252a1 1 0 0 0 1.328 1.496zm-4.678 1.417 2-12a1 1 0 1 0-1.972-.329l-2 12a1 1 0 1 0 1.972.329z"
              />
            </svg>
          </a>
        </section>
      </section>
      <section
        class="c3"
        title="Article Search"
      >
        <a
          class="c4"
          href="https://ecioferraz.github.io/mettzer-article-search/"
          target="_blank"
        >
          <img
            alt="Imagem do projeto Article Search"
            class="c5"
            src="/@fs/home/ecio/Documentos/trybe/Portfólio/ecioferraz.github.io/public/assets/images/projects/articleSearch.png"
          />
        </a>
        <section
          class="c6"
        >
          <h4
            class="c7"
          >
            Article Search
          </h4>
          <a
            class="c4"
            href="https://github.com/ecioferraz/mettzer-article-search"
            target="_blank"
          >
            <svg
              class="c8"
              fill="currentColor"
              focusable="false"
              role="img"
              viewBox="0 0 20 20"
              xmlns="http://www.w3.org/2000/svg"
            >
              <title>
                GitHub
              </title>
              <path
                d="M5.719 14.75a.997.997 0 0 1-.664-.252L-.005 10l5.341-4.748a1 1 0 0 1 1.328 1.495L3.005 10l3.378 3.002a1 1 0 0 1-.664 1.748zm8.945-.002L20.005 10l-5.06-4.498a.999.999 0 1 0-1.328 1.495L16.995 10l-3.659 3.252a1 1 0 0 0 1.328 1.496zm-4.678 1.417 2-12a1 1 0 1 0-1.972-.329l-2 12a1 1 0 1 0 1.972.329z"
              />
            </svg>
          </a>
        </section>
      </section>
      <section
        class="c3"
        title="TrybeWallet"
      >
        <a
          class="c4"
          href="https://ecioferraz.github.io/trybe-trybewallet-project/"
          target="_blank"
        >
          <img
            alt="Imagem do projeto TrybeWallet"
            class="c5"
            src="/@fs/home/ecio/Documentos/trybe/Portfólio/ecioferraz.github.io/public/assets/images/projects/trybewallet.png"
          />
        </a>
        <section
          class="c6"
        >
          <h4
            class="c7"
          >
            TrybeWallet
          </h4>
          <a
            class="c4"
            href="https://github.com/ecioferraz/trybe-trybewallet-project"
            target="_blank"
          >
            <svg
              class="c8"
              fill="currentColor"
              focusable="false"
              role="img"
              viewBox="0 0 20 20"
              xmlns="http://www.w3.org/2000/svg"
            >
              <title>
                GitHub
              </title>
              <path
                d="M5.719 14.75a.997.997 0 0 1-.664-.252L-.005 10l5.341-4.748a1 1 0 0 1 1.328 1.495L3.005 10l3.378 3.002a1 1 0 0 1-.664 1.748zm8.945-.002L20.005 10l-5.06-4.498a.999.999 0 1 0-1.328 1.495L16.995 10l-3.659 3.252a1 1 0 0 0 1.328 1.496zm-4.678 1.417 2-12a1 1 0 1 0-1.972-.329l-2 12a1 1 0 1 0 1.972.329z"
              />
            </svg>
          </a>
        </section>
      </section>
      <section
        class="c3"
        title="Trivia"
      >
        <a
          class="c4"
          href="https://ecioferraz.github.io/trybe-trivia-project/"
          target="_blank"
        >
          <img
            alt="Imagem do projeto Trivia"
            class="c5"
            src="/@fs/home/ecio/Documentos/trybe/Portfólio/ecioferraz.github.io/public/assets/images/projects/trivia.png"
          />
        </a>
        <section
          class="c6"
        >
          <h4
            class="c7"
          >
            Trivia
          </h4>
          <a
            class="c4"
            href="https://github.com/ecioferraz/trybe-trivia-project"
            target="_blank"
          >
            <svg
              class="c8"
              fill="currentColor"
              focusable="false"
              role="img"
              viewBox="0 0 20 20"
              xmlns="http://www.w3.org/2000/svg"
            >
              <title>
                GitHub
              </title>
              <path
                d="M5.719 14.75a.997.997 0 0 1-.664-.252L-.005 10l5.341-4.748a1 1 0 0 1 1.328 1.495L3.005 10l3.378 3.002a1 1 0 0 1-.664 1.748zm8.945-.002L20.005 10l-5.06-4.498a.999.999 0 1 0-1.328 1.495L16.995 10l-3.659 3.252a1 1 0 0 0 1.328 1.496zm-4.678 1.417 2-12a1 1 0 1 0-1.972-.329l-2 12a1 1 0 1 0 1.972.329z"
              />
            </svg>
          </a>
        </section>
      </section>
      <section
        class="c3"
        title="Star Wars: Planet Search"
      >
        <a
          class="c4"
          href="https://ecioferraz.github.io/trybe-star-wars-project/"
          target="_blank"
        >
          <img
            alt="Imagem do projeto Star Wars: Planet Search"
            class="c5"
            src="/@fs/home/ecio/Documentos/trybe/Portfólio/ecioferraz.github.io/public/assets/images/projects/star-wars.png"
          />
        </a>
        <section
          class="c6"
        >
          <h4
            class="c7"
          >
            Star Wars: Planet Search
          </h4>
          <a
            class="c4"
            href="https://github.com/ecioferraz/trybe-star-wars-project"
            target="_blank"
          >
            <svg
              class="c8"
              fill="currentColor"
              focusable="false"
              role="img"
              viewBox="0 0 20 20"
              xmlns="http://www.w3.org/2000/svg"
            >
              <title>
                GitHub
              </title>
              <path
                d="M5.719 14.75a.997.997 0 0 1-.664-.252L-.005 10l5.341-4.748a1 1 0 0 1 1.328 1.495L3.005 10l3.378 3.002a1 1 0 0 1-.664 1.748zm8.945-.002L20.005 10l-5.06-4.498a.999.999 0 1 0-1.328 1.495L16.995 10l-3.659 3.252a1 1 0 0 0 1.328 1.496zm-4.678 1.417 2-12a1 1 0 1 0-1.972-.329l-2 12a1 1 0 1 0 1.972.329z"
              />
            </svg>
          </a>
        </section>
      </section>
      <section
        class="c3"
        title="Pixel Art"
      >
        <a
          class="c4"
          href="https://ecioferraz.github.io/trybe-pixel-art-project/"
          target="_blank"
        >
          <img
            alt="Imagem do projeto Pixel Art"
            class="c5"
            src="/@fs/home/ecio/Documentos/trybe/Portfólio/ecioferraz.github.io/public/assets/images/projects/pixel-art.png"
          />
        </a>
        <section
          class="c6"
        >
          <h4
            class="c7"
          >
            Pixel Art
          </h4>
          <a
            class="c4"
            href="https://github.com/ecioferraz/trybe-pixel-art-project"
            target="_blank"
          >
            <svg
              class="c8"
              fill="currentColor"
              focusable="false"
              role="img"
              viewBox="0 0 20 20"
              xmlns="http://www.w3.org/2000/svg"
            >
              <title>
                GitHub
              </title>
              <path
                d="M5.719 14.75a.997.997 0 0 1-.664-.252L-.005 10l5.341-4.748a1 1 0 0 1 1.328 1.495L3.005 10l3.378 3.002a1 1 0 0 1-.664 1.748zm8.945-.002L20.005 10l-5.06-4.498a.999.999 0 1 0-1.328 1.495L16.995 10l-3.659 3.252a1 1 0 0 0 1.328 1.496zm-4.678 1.417 2-12a1 1 0 1 0-1.972-.329l-2 12a1 1 0 1 0 1.972.329z"
              />
            </svg>
          </a>
        </section>
      </section>
    </section>
  </main>
</div>
`;
